Source/Fuse.Scripting.JavaScript/JavaScript.Model.uno

```diff
@@ -0,0 +1,254 @@
+		{
+			public IExpression Model;
+			public NameTable NameTable;
+		}
+		
+		//Requires a NameTable that will be set first.
+        [UXAttachedPropertySetter("JavaScript.Model"), UXNameScope, UXAuxNameTable("ModelNameTable")]
+        public static void SetModel(Visual v, IExpression model)
+        {
+			var md = v.Properties.Get( _modelHandle ) as ModelData;
+			if (md == null)
+			{
+				md = new ModelData{ Model = model };
+				v.Properties.Set( _modelHandle, md );
+			}
+			else
+			{
+				md.Model = model;
+			}
+			
+			Complete( md, v );
+		}
+		
+		static void Complete( ModelData md, Visual v )
+		{
+			v.RemoveAllChildren<ModelJavaScript>();
+			
+			//avoid creating without the NameTable as unfortunately UX will set the Model prior to the NameTable
+			if (md.NameTable == null || md.Model == null)
+				return;
+				
+			var parsed = ModelJavaScript.ParseModelExpression(md.Model, md.NameTable);
+			v.Children.Add( new ModelJavaScript(parsed, md.NameTable, null) );
+        }
+
+        static PropertyHandle _modelHandle = Properties.CreateHandle();
+        
+        [UXAttachedPropertySetterAttribute("ModelNameTable")]
+        public static void SetModelNameTable(Visual v, NameTable nt)
+        {
+			var md = v.Properties.Get( _modelHandle ) as ModelData;
+			if (md == null)
+			{
+				md = new ModelData{ NameTable = nt };
+				v.Properties.Set( _modelHandle, md );
+			}
+			else
+			{
+				md.NameTable = nt;
+			}
+			
+			Complete( md, v );
+        }
+
+        //TODO: This should probably be JavaScript.Model, Preview would need to be adjusted as well
+        [UXAttachedPropertySetter("Model"), UXNameScope]
+        public static void SetAppModel(IApp app, IExpression model)
+        {
+			app.Root.RemoveAllChildren<ModelJavaScript>();
+
+			var _appModel = ModelJavaScript.CreateFromPreviewState(app.Root, model);
+			app.Root.Children.Add(_appModel);
+        }
+	}
+	
+	class ModelJavaScript : JavaScript, IPreviewStateSaver
+	{
+		internal class ParsedModelExpression
+		{
+			public IExpression Source;
+			public string ModuleName;
+			public string ClassName;
+			public List<string> Args = new List<string>();
+			public List<Dependency> Dependencies = new List<Dependency>();
+			
+			public string ArgString
+			{
+				get
+				{
+					var str = "";
+					for (int i=0; i < Args.Count; ++i)
+						str += ", " + Args[i];
+					return str;
+				}
+			}
+			
+			public bool CompatibleTo( ParsedModelExpression o )
+			{
+				//there is no way to migrate these now as they might refer to tree objects, thus reject entirely
+				if (Args.Count != 0 || o.Args.Count != 0 ||	
+					Dependencies.Count != 0 || o.Dependencies.Count != 0)
+					return false;
+					
+				return o.ModuleName == ModuleName &&
+					o.ClassName == ClassName;
+			}
+		}
+		
+		internal static ParsedModelExpression ParseModelExpression(IExpression exp, NameTable nt)
+        {
+            if (exp is Data) {
+                var className = ((Data)exp).Key;
```

Comment:
First of all, please no Egyptian braces.

Second, please drop the double-cast, instead do:
```Uno
var data = exp as Data;
if (data !=null)
{
	var className = data.Key;
	return [...]
}

var divide = exp as Divide;
if (divide != null)
[...]
```